app/models/request/XSDFile.scala

```diff
@@ -17,27 +17,34 @@
 package models.request
 
 sealed trait XSDFile {
-  val filePath: String
-  val label: String
+  val FilePath: String
+  val Label: String
 }
 
+sealed trait ArrivalMessage
+
 object XSDFile {
-  val values = Seq(ArrivalNotificationXSD, DepartureDeclarationXSD, UnloadingRemarksXSD)
-  val map = values.map(xsd => xsd.label -> xsd).toMap
-  val supportedMessageTypes = map.filterKeys(x => (x != ArrivalNotificationXSD.label) && (x != DepartureDeclarationXSD.label))
+  object Arrival {
+    val Definitions = Seq(ArrivalNotificationXSD, UnloadingRemarksXSD)
+    val DefinitionsMap = Definitions.map(xsd => xsd.Label -> xsd).toMap
+    val SupportedMessages: Map[String, XSDFile] = DefinitionsMap.filter {
+      case (key: String, value: XSDFile) =>
+        value.isInstanceOf[ArrivalMessage]
+    }
+  }
 }
 
 object ArrivalNotificationXSD extends XSDFile {
```

Comment:
Should ArrivalNotificationXSD also have the ArrivalMessage trait?

Reply:
> Should ArrivalNotificationXSD also have the ArrivalMessage trait?

No as we don't want to allow the user to send an IE007 to the /messages endpoint.